src/views/virtualmachines/tree/VirtualMachineTreeView.tsx

```diff
@@ -0,0 +1,136 @@
+import React, { FC, MouseEvent, useMemo } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom-v5-compat';
+
+import VirtualMachineModel from '@kubevirt-ui/kubevirt-api/console/models/VirtualMachineModel';
+import { useQueryParamsMethods } from '@kubevirt-utils/components/ListPageFilter/hooks/useQueryParamsMethods';
+import { convertResourceArrayToMap, getResourceUrl } from '@kubevirt-utils/resources/shared';
+import { FilterValue, useActiveNamespace } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  Drawer,
+  DrawerContent,
+  DrawerContentBody,
+  DrawerPanelBody,
+  DrawerPanelContent,
+  TreeView,
+  TreeViewDataItem,
+} from '@patternfly/react-core';
+import { TEXT_FILTER_LABELS_ID } from '@virtualmachines/list/hooks/constants';
+
+import TreeViewToolbar from './components/TreeViewToolbar';
+import { useHideNamespaceBar } from './hooks/useHideNamespaceBar';
+import { useSyncClicksEffects } from './hooks/useSyncClicksEffects';
+import { useTreeViewData } from './hooks/useTreeViewData';
+import { useTreeViewSearch } from './hooks/useTreeViewSearch';
+import {
+  FOLDER_SELECTOR_PREFIX,
+  PROJECT_SELECTOR_PREFIX,
+  TREE_VIEW_PANEL_ID,
+  VM_FOLDER_LABEL,
+} from './utils/constants';
+import {
+  createTreeViewData,
+  selectedTreeItem,
+  setSelectedTreeItem,
+  treeViewOpen,
+} from './utils/utils';
+
+import './VirtualMachineTreeView.scss';
+
+type VirtualMachineTreeViewProps = {
+  onFilterChange?: (type: string, value: FilterValue) => void;
+};
+
+const VirtualMachineTreeView: FC<VirtualMachineTreeViewProps> = ({ children, onFilterChange }) => {
+  const [activeNamespace] = useActiveNamespace();
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const { setOrRemoveQueryArgument } = useQueryParamsMethods();
+  const { isAdmin, loaded, loadError, projectNames, vms } = useTreeViewData();
+  const vmsMapper = useMemo(() => convertResourceArrayToMap(vms, true), [vms]);
+
+  const treeData = useMemo(
+    () => createTreeViewData(projectNames, vms, activeNamespace, isAdmin, location.pathname),
+    [projectNames, vms, activeNamespace, isAdmin, location.pathname],
+  );
+
+  const { filteredItems, onSearch, setShowAll, showAll } = useTreeViewSearch(treeData);
+
+  useSyncClicksEffects(activeNamespace, loaded, location);
```

Comment:
do we still need it? even after hiding the ns bar? activeNamespace isn't enough?

Reply:
This is still needed to be in sync with vm list click not from treeview